crates/nu-engine/src/script.rs

```diff
@@ -113,7 +104,7 @@ pub async fn process_script(
                         .unwrap_or(true)
                     && canonicalize(ctx.shell_manager.path(), name).is_ok()
                     && Path::new(&name).is_dir()
-                    && !crate::commands::classified::external::did_find_command(&name)
+                    && !ctx.host.lock().is_external_cmd(&name)
```

Comment:
There was a recent PR that came in that fixed some of the locking. Might be worth double-checking it to make sure this still applies.